components/selection/selection-select-all-pages.js

```diff
@@ -40,7 +40,10 @@ class SelectAllPages extends FocusMixin(LocalizeCoreElement(SelectionObserverMix
 	}
 
 	_handleClick() {
-		if (this._provider) this._provider.setSelectionForAll(true, true);
+		if (!this._provider) return;
+
+		this._provider.setSelectionForAll(true, true);
+		this._provider._focusSelectAll();
```

Comment:
We delegate this task to the provider to locate and focus on the `d2l-selection-select-all`. We don't rely on an event because there may not be an ancestor in the event path that knows how to locate the select-all. The selection provider is in the best position to identify the select-all.